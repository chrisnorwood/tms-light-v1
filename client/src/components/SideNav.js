import React from 'react'
import { NavLink } from 'react-router-dom'
import { FaBriefcase, FaTruck, FaAddressBook, FaTruckLoading, FaSignOutAlt, FaHome } from 'react-icons/fa'

const SideNav = () => {
  return (
    <div className='fixed w-56 min-h-screen text-grey-darkest font-bold shadow-2xl bg-grey-light'>
      <div className='py-4 text-center text-2xl border-b border-grey-dark'>
        TMS
      </div>
      <ul className='sidebar-nav'>
        <li>
          <NavLink
            exact
            to='/app'
            className='nav-link'
          >
            <FaHome className='menu-icon'/>
            <span className='ml-8'>Dashboard</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            exact
            to='/app/loads'
            className='nav-link'
          >
            <FaTruckLoading className='menu-icon'/>
            <span className='ml-8'>Loads</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            exact
            to='/app/contacts'
            className='nav-link'
          >
            <FaAddressBook className='menu-icon'/>
            <span className='ml-8'>Contacts</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            exact
            to='/app/shippers'
            className='nav-link'
          >
            <FaBriefcase className='menu-icon'/>
            <span className='ml-8'>Shippers</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            exact
            to='/app/carriers'
            className='nav-link'
          >
            <FaTruck className='menu-icon'/>
            <span className='ml-8'>Carriers</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            exact
            to='/logout'
            className='nav-link'
          >
            <FaSignOutAlt className='menu-icon'/>
            <span className='ml-8'>Logout</span>
          </NavLink>
        </li>
      </ul>
    </div>
  )
}

export default SideNav