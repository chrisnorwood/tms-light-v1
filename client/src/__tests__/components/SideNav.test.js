import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import SideNav from '../../components/SideNav'

afterEach(cleanup)

it('renders the SideNav', () => {
  const { asFragment } = render(
    // Include Router to avoid rendering NavLink outside of Router errors
    <Router>
      <SideNav />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
})
