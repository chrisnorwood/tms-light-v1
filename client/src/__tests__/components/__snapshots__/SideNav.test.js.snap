// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SideNav 1`] = `
<DocumentFragment>
  <div
    class="fixed w-56 min-h-screen text-grey-darkest font-bold shadow-2xl bg-grey-light"
  >
    <div
      class="h-2 bg-primary"
    />
    <div
      class="py-4 text-center text-2xl border-b border-grey-dark text-primary"
    >
      TMS
    </div>
    <ul
      class="sidebar-nav"
    >
      <li>
        <a
          class="nav-link"
          href="/app/loads"
        >
          <svg
            class="menu-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50.2 375.6c2.3 8.5 11.1 13.6 19.6 11.3l216.4-58c8.5-2.3 13.6-11.1 11.3-19.6l-49.7-185.5c-2.3-8.5-11.1-13.6-19.6-11.3L151 133.3l24.8 92.7-61.8 16.5-24.8-92.7-77.3 20.7C3.4 172.8-1.7 181.6.6 190.1l49.6 185.5zM384 0c-17.7 0-32 14.3-32 32v323.6L5.9 450c-4.3 1.2-6.8 5.6-5.6 9.8l12.6 46.3c1.2 4.3 5.6 6.8 9.8 5.6l393.7-107.4C418.8 464.1 467.6 512 528 512c61.9 0 112-50.1 112-112V0H384zm144 448c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
            />
          </svg>
          <span
            class="ml-8"
          >
            Loads
          </span>
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/app/contacts"
        >
          <svg
            class="menu-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M436 160c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h320c26.5 0 48-21.5 48-48v-48h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-20v-64h20zm-228-32c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H118.4C106 384 96 375.4 96 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
            />
          </svg>
          <span
            class="ml-8"
          >
            Contacts
          </span>
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/app/shippers"
        >
          <svg
            class="menu-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
            />
          </svg>
          <span
            class="ml-8"
          >
            Shippers
          </span>
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/app/carriers"
        >
          <svg
            class="menu-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h16c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
            />
          </svg>
          <span
            class="ml-8"
          >
            Carriers
          </span>
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/logout"
        >
          <svg
            class="menu-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
            />
          </svg>
          <span
            class="ml-8"
          >
            Logout
          </span>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
